AppTests/App/Dependencies/ImmuniTemporaryExposureKeyProviderTests.swift

```diff
@@ -0,0 +1,116 @@
+
+    let promise = keyProvider.getMissingChunksIndexes(latestKnownChunkIndex: nil)
+
+    expectToEventually(promise.isPending == false)
+
+    let chunks = try XCTUnwrap(promise.result)
+
+    XCTAssertEqual(chunks.first, 50)
+    XCTAssertEqual(chunks.last, 60)
+  }
+
+  func testReturnCorrectValuesWithLittleChunks() throws {
+    let mockedExecutor = MockRequestExecutor(mockedResult: .success(KeysIndex(oldest: 1, newest: 3)))
+    let networkManager = NetworkManager()
+    networkManager.start(with: .init(requestExecutor: mockedExecutor, now: { Date() }))
+
+    let keyProvider = ImmuniTemporaryExposureKeyProvider(networkManager: networkManager, fileStorage: MockFileStorage())
+
+    let promise = keyProvider.getMissingChunksIndexes(latestKnownChunkIndex: nil)
+
+    expectToEventually(promise.isPending == false)
+
+    let chunks = try XCTUnwrap(promise.result)
+
+    XCTAssertEqual(chunks.first, 1)
+    XCTAssertEqual(chunks.last, 3)
+  }
+
+  func testReturnCorrectValuesWithKnownChunk() throws {
+    let mockedExecutor = MockRequestExecutor(mockedResult: .success(KeysIndex(oldest: 50, newest: 60)))
+    let networkManager = NetworkManager()
+    networkManager.start(with: .init(requestExecutor: mockedExecutor, now: { Date() }))
+
+    let keyProvider = ImmuniTemporaryExposureKeyProvider(networkManager: networkManager, fileStorage: MockFileStorage())
+
+    let promise = keyProvider.getMissingChunksIndexes(latestKnownChunkIndex: 55)
+
+    expectToEventually(promise.isPending == false)
+
+    let chunks = try XCTUnwrap(promise.result)
+
+    XCTAssertEqual(chunks.first, 56)
+    XCTAssertEqual(chunks.last, 60)
+  }
+
+  func testReturnCorrectValuesOnRateLimit() throws {
+    let mockedExecutor = MockRequestExecutor(mockedResult: .success(KeysIndex(oldest: 50, newest: 80)))
+    let networkManager = NetworkManager()
+    networkManager.start(with: .init(requestExecutor: mockedExecutor, now: { Date() }))
+
+    let keyProvider = ImmuniTemporaryExposureKeyProvider(networkManager: networkManager, fileStorage: MockFileStorage())
+
+    let promise = keyProvider.getMissingChunksIndexes(latestKnownChunkIndex: 55)
+
+    expectToEventually(promise.isPending == false)
+
+    let chunks = try XCTUnwrap(promise.result)
+
+    XCTAssertEqual(chunks.first, 66) // 80 minus the Apple rate limit
+    XCTAssertEqual(chunks.last, 80)
+  }
+
+  func testReturnCorrectValuesOnNoChunks() throws {
+    let mockedExecutor = MockRequestExecutor(mockedResult: .success(KeysIndex(oldest: 50, newest: 80)))
+    let networkManager = NetworkManager()
+    networkManager.start(with: .init(requestExecutor: mockedExecutor, now: { Date() }))
+
+    let keyProvider = ImmuniTemporaryExposureKeyProvider(networkManager: networkManager, fileStorage: MockFileStorage())
+
+    let promise = keyProvider.getMissingChunksIndexes(latestKnownChunkIndex: 80)
+
+    expectToEventually(promise.isPending == false)
+
+    let chunks = try XCTUnwrap(promise.result)
+    
```

Comment:
<!--
  0 failure: 
  1 warning:  swiftlint (0.39.2...
  
  
  DangerID: danger-id-Danger;
  File: AppTests/App/Dependencies/ImmuniTemporaryExposureKeyProviderTests.swift;
  Line: 103;
-->

- :warning: swiftlint (0.39.2): Lines should not have trailing whitespace. (trailing_whitespace)